packages/jest-config/src/normalize.js

```diff
@@ -145,21 +145,25 @@ const normalizeCollectCoverageOnlyFrom = (
 };
 
 const normalizeCollectCoverageFrom = (options: InitialOptions, key: string) => {
-  let value;
-  if (!options[key]) {
-    value = [];
+  let value = options[key];
+
+  if (!value) {
+    return [];
   }
 
-  if (!Array.isArray(options[key])) {
+  if (!Array.isArray(value)) {
     try {
-      value = JSON.parse(options[key]);
-    } catch (e) {}
-
-    Array.isArray(value) || (value = [options[key]]);
-  } else {
-    value = options[key];
+      const parsed = JSON.parse(options[key]);
+      Array.isArray(parsed) && (value = parsed);
```

Comment:
ugh. Can we just use an `if` here?

```js
if (Array.isArray(parsed)) {
  value = parsed;
}
```

Reply:
this is also the wrong logic, isn't it? I think the `JSON.parse` should be by its own in the `try`

Reply:
Yes, we can! /obamamode off

Reply:
I think the best would be to keep the logic you've added to line 163, and revert the other changes (feel free to exchange the `||` for an `if`, though :P)

Reply:
Nah, the logic is solid. It's just that before my changing it was inside a try block assigning to the value and using the catch block to revert the value if it went wrong. This actually allowed an object to be passed though. For example:

```
normalizeCollectCoverageFrom({
  collectCoverageFrom: {foo: 'bar'}
})
```
Now, this is wrong of course, as it's supposed to take an array. As such, it would have entered the original if-statement that checked it to be an array. And since it's not an array, it would be parsed by `JSON.parse`, which would in turn throw an error as it's being given an object. The result would be the catch block assigning the original object back to the value and where you would expect an array, you suddenly have an object.

As such, the logic after the refactoring will always assure the result is an array. So while I could return the original code, I would rather not.

Reply:
Reading through the code again, if the JSON string is not a representation of an array, this would still have an object as the value. I'll return the original 😆